### ISSUE TITLE
SRP0014 false positive

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/SqlServer.Rules#718]
The target for the update is not a table,  but a table variable!

```sql
DECLARE @body NVARCHAR(MAX) = N'
{
	"data" : [
		{
			"originalId" : "abc1",
			"currentId": "abc2"
		},
		{
			"originalId" : "cde1",
			"currentId": "cde2"
		}
	]
}
';
DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM @someTable dst
INNER JOIN OPENJSON(@body, '$.data') WITH (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
) bodyItems
ON bodyItems.originalId = dst.originalId;

```

@niphlod FYI

**Comment by ErikEJ** [comment]
@niphlod As stated earlier, table variables should generally be avoided
